vms/proposervm/block.go

```diff
@@ -334,3 +303,75 @@ func verifyIsNotOracleBlock(ctx context.Context, b snowman.Block) error {
 		return err
 	}
 }
+
+func (p *postForkCommonComponents) verifyBlockDelay(
+	ctx context.Context,
+	parentTimestamp time.Time,
+	parentPChainHeight uint64,
+	blk *postForkBlock,
+) (time.Duration, error) {
+	var (
+		blkTimestamp = blk.Timestamp()
+		childHeight  = blk.Height()
+		proposerID   = blk.Proposer()
+	)
+	minDelay, err := p.vm.Windower.Delay(ctx, childHeight, parentPChainHeight, proposerID, proposer.MaxVerifyWindows)
+	if err != nil {
+		return 0, err
+	}
+
+	delay := blkTimestamp.Sub(parentTimestamp)
+	if delay < minDelay {
+		return 0, errProposerWindowNotStarted
+	}
+
+	return delay, nil
+}
+
+func (p *postForkCommonComponents) shouldBuildBlock(
+	ctx context.Context,
+	parentID ids.ID,
+	parentTimestamp time.Time,
+	parentPChainHeight uint64,
+	newTimestamp time.Time,
+) (bool, error) {
```

Comment:
Can we make the name of this function a bit more clear? Looking at it I would expect the `bool` to return `false, nil` if `delay < minDelay` and `true, nil` if `delay >= minDelay` - but really the `error` is checking if the block should be built and the `bool` is flagging if it should be signed or unsigned